pkg/controllers/user/rbac/legacy_rt_cleaner.go

```diff
@@ -49,15 +53,14 @@ func (p *rtCleaner) removeStuckFinalizer(obj *v3.RoleTemplate) (*v3.RoleTemplate
 	for i := len(finalizers) - 1; i >= 0; i-- {
 		f := finalizers[i]
 		if strings.HasPrefix(f, lifecycle.ScopedFinalizerKey) {
-			finalizers = append(finalizers[:i], finalizers[i+1:]...)
-			modified = true
+			s := strings.Split(f, "_")
+			// if cluster was not reported, its a deleted cluster and will cause the finalizer to hang in the future
+			if _, err = p.m.clusterLister.Get("", s[1]); errors.IsNotFound(err) {
```

Comment:
Are we guaranteed every finalizer always has underscore ? If not then `s[1]` could error out.
Might also be a more descriptive var name here than `s`. The `f` var works because its short for finalizer but what does s mean.

Reply:
For cluster scoped resources we have a defined format of `clusterscoped.controller.cattle.io/cluster-roletemplate-<clusterName>"`

Reply:
https://github.com/golang/go/wiki/CodeReviewComments#variable-names